### ISSUE TITLE
`ssh` and related macros should use fully qualified `clojure.core/identity`

### ISSUE THREAD

**Issue by tomconnors** [epiccastle/spire#75]
Passing in the `:identity` option to `ssh` as a local called `identity` causes an issue because the macro uses `identity` without qualifying as `clojure.core/identity`.

This doesn't work:
``` clojure
(let [{:keys [user host identity]} args]
  (spire/ssh
   {:username user
    :hostname host
    :identity identity}
   (spire/get-fact)))
```

But this does:
``` clojure
(let [{:keys [user host]} args]
  (spire/ssh
   {:username user
    :hostname host
    :identity (:identity args)}
   (spire/get-fact)))
```

**Comment by retrogradeorbit** [comment]
Very strange... 
```
(macroexpand-1 '(ssh {:username "user" :hostname "host" :identity "ident"} body))
;; => (clojure.core/let [host-config__30081__auto__ (spire.ssh/host-description-to-host-config {:username "user", :hostname "host", :identity "ident"})] (try (clojure.core/let [conn__30082__auto__ (spire.transport/open-connection host-config__30081__auto__)] (spire.context/binding* [spire.state/host-config host-config__30081__auto__ spire.state/connection conn__30082__auto__ spire.state/shell-context {:stdin-fn clojure.core/identity, :exec :ssh, :priveleges :normal, :shell-fn clojure.core/identity, :exec-fn spire.ssh/ssh-exec}] (spire.facts/update-facts!) (do body))) (finally (spire.transport/close-connection host-config__30081__auto__))))
```

will investigate

**Comment by retrogradeorbit** [comment]
This works for me:

```clojure
(ns test
  (:require [spire.modules :as spire]))

(def args
  {:user "root"
   :host "epiccastle.io"
   :identity "/home/crispin/.ssh/id_rsa"})

(let [{:keys [user host identity]} args]
  (spire/ssh
   {:username user
    :hostname host
    :identity identity}
   (spire/get-fact [:system :os])))
```

```
$ spire test-issue-75.clj 
Enter Passphrase for /home/crispin/.ssh/id_rsa: 
:linux
```

Can you share a little more context around the running of that failing code?

**Comment by retrogradeorbit** [comment]
All good. I just recreated writing a test for #76 

```
Exception in thread "main" clojure.lang.ExceptionInfo: Cannot call "/home/crispin/.ssh/id_rsa" as a function. [at test-issue-76.clj, line 10, column 3] {:type :sci/error, :line 10, :column 3, :message "Cannot call \"/home/crispin/.ssh/id_rsa\" as a function. [at test-issue-76.clj, line 10, column 3]"}
	at sci.impl.utils$rethrow_with_location_of_node.invokeStatic(utils.cljc:90)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:517)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:557)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:534)
	at sci.impl.interpreter$eval_let.invokeStatic(interpreter.cljc:83)
	at sci.impl.interpreter$eval_let.doInvoke(interpreter.cljc:59)
	at clojure.lang.RestFn.applyTo(RestFn.java:142)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at sci.impl.interpreter$eval_special_call.invokeStatic(interpreter.cljc:481)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:508)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:557)
	at sci.impl.interpreter$eval_form.invokeStatic(interpreter.cljc:589)
	at sci.impl.interpreter$eval_string_STAR_.invokeStatic(interpreter.cljc:602)
	at sci.core$eval_string_STAR_.invokeStatic(core.cljc:189)
	at spire.eval$evaluate.invokeStatic(eval.clj:87)
	at spire.core$_main.invokeStatic(core.clj:80)
	at spire.core$_main.doInvoke(core.clj:57)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at spire.core.main(Unknown Source)
Caused by: java.lang.Exception: Cannot call "/home/crispin/.ssh/id_rsa" as a function.
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:517)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:557)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:534)
	at sci.impl.interpreter$fn_call.invokeStatic(interpreter.cljc:467)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:517)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:557)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:534)
	at sci.impl.interpreter$eval_do_STAR_$fn__5444.invoke(interpreter.cljc:424)
	at sci.impl.interpreter$eval_do_STAR_.invokeStatic(interpreter.cljc:424)
	at sci.impl.interpreter$eval_do.invokeStatic(interpreter.cljc:433)
	at sci.impl.interpreter$eval_special_call.invokeStatic(interpreter.cljc:500)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:508)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:557)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:534)
	at sci.impl.interpreter$eval_do_STAR_$fn__5444.invoke(interpreter.cljc:424)
	at sci.impl.interpreter$eval_do_STAR_.invokeStatic(interpreter.cljc:424)
	at sci.impl.interpreter$eval_do.invokeStatic(interpreter.cljc:433)
	at sci.impl.interpreter$eval_special_call.invokeStatic(interpreter.cljc:500)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:508)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:557)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:534)
	at sci.impl.interpreter$eval_do_STAR_$fn__5444.invoke(interpreter.cljc:424)
	at sci.impl.interpreter$eval_do_STAR_.invokeStatic(interpreter.cljc:424)
	at sci.impl.interpreter$eval_do.invokeStatic(interpreter.cljc:433)
	at sci.impl.interpreter$eval_special_call.invokeStatic(interpreter.cljc:500)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:508)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:557)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:534)
	at sci.impl.interpreter$eval_try.invokeStatic(interpreter.cljc:289)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:557)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:534)
	at sci.impl.interpreter$eval_let.invokeStatic(interpreter.cljc:83)
	at sci.impl.interpreter$eval_let.doInvoke(interpreter.cljc:59)
	at clojure.lang.RestFn.applyTo(RestFn.java:142)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at sci.impl.interpreter$eval_special_call.invokeStatic(interpreter.cljc:481)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:508)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:557)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:534)
	at sci.impl.interpreter$eval_let.invokeStatic(interpreter.cljc:83)
	at sci.impl.interpreter$eval_let.doInvoke(interpreter.cljc:59)
	at clojure.lang.RestFn.applyTo(RestFn.java:142)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at sci.impl.interpreter$eval_special_call.invokeStatic(interpreter.cljc:481)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:508)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:557)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:534)
	at sci.impl.interpreter$eval_do_STAR_$fn__5444.invoke(interpreter.cljc:424)
	at sci.impl.interpreter$eval_do_STAR_.invokeStatic(interpreter.cljc:424)
	at sci.impl.interpreter$eval_do.invokeStatic(interpreter.cljc:433)
	at sci.impl.interpreter$eval_special_call.invokeStatic(interpreter.cljc:500)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:508)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:557)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:534)
	at sci.impl.interpreter$eval_try.invokeStatic(interpreter.cljc:289)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:557)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:534)
	at sci.impl.interpreter$eval_let.invokeStatic(interpreter.cljc:83)
	at sci.impl.interpreter$eval_let.doInvoke(interpreter.cljc:59)
	at clojure.lang.RestFn.applyTo(RestFn.java:142)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at sci.impl.interpreter$eval_special_call.invokeStatic(interpreter.cljc:481)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:508)
	... 17 more

```

**Comment by retrogradeorbit** [comment]
fixed in 0.1.0-alpha.13
